packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -262,6 +262,16 @@ export const BlocksDropdown = () => {
     }
   };
 
+  React.useEffect(() => {
+    const selectedBlocks = Object.keys(blocks).filter((key) => {
+      return isBlockActive(editor, blocks[key].matchNode);
+    });
+
+    if (selectedBlocks.length === 1) {
+      setBlockSelected(selectedBlocks[0]);
```

Comment:
I think just setting anchor node should get rid of finding all selected blocks and just set first selected, its not very clear

Reply:
@simotae14 I agree with @madhurisandbhor, relying on the anchor of the selection should simplify the logic, that's what I meant the other day on the slack thread.

My guess would be that the change needs to happen inside of `isBlockActive`, updating the `at` param of `Editor.nodes`

Reply:
I have updated the logic based on your suggestions (thanks a lot @madhurisandbhor for the help), let me know if you agree with the solution, I need to work and fix a unit test that is not working as intended.